apps/gdalbuildvrt_lib.cpp

```diff
@@ -849,6 +897,37 @@ std::string VRTBuilder::AnalyseRaster(GDALDatasetH hDS,
                 const int nSelBand = panSelectedBandList[j];
                 CPLAssert(nSelBand >= 1 && nSelBand <= _nBands);
                 GDALRasterBand *poBand = poDS->GetRasterBand(nSelBand);
+                // In normal mode we only preserve description if identical across
+                if (asBandProperties[j].osDescription !=
+                    poBand->GetDescription())
+                {
+                    asBandProperties[j].osDescription = "";
+                }
+                // same for metadata
+                char **papszMD = poBand->GetMetadata();
```

Comment:
I believe this test is incomplete, and will miss the case where the first dataset has a key that is absent in following datasets.